modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -344,5 +345,6 @@ export const insertHtmlAtCaret = (editor: Editor, value: string, details: Insert
   moveSelectionToMarker(editor, dom.get('mce_marker'));
   unmarkFragmentElements(editor.getBody());
   trimBrsFromTableCell(dom, selection.getStart());
+  TransparentElements.update(editor.schema, editor.getBody(), true);
```

Comment:
Do we have to do this on the entire body? Couldn't we just do the inserted content, or perhaps the parent of the inserted content?

Reply:
Did some optimization now it's a bit tricky since it might need to update nodes up to the root.
Say you insert `<p>bar</p>` in `<div><a href="#">a|b</a></div>` then it should update data-mce-block state so that it becomes `<div><a href="#" data-mce-block="true">a<p>bar|</p>b</a></div>` so I changed it a bit so it goes down to the element just above the root and updates that if it find one otherwise it updates all elements.

Reply:
Could be more complex structures as well say you insert `<a href="#><p>foo</p></a><i><b>bar</b></i>` in `<div>baz|</div>` then it should become `<div>baz<a href="#" data-mce-block="true"><p>foo</p></a><i><b>bar|</b></i></div>` there is a risk that we miss things unless we cast a wide net. So checking the parent wouldn't be enough in this case.

Reply:
Added some more tests around this.